src/Interpreters/Session.cpp

```diff
@@ -401,13 +406,20 @@ ContextMutablePtr Session::makeSessionContext()
     prepared_client_info.reset();
 
     /// Set user information for the new context: current profiles, roles, access rights.
+    /// TCP handler quota is received after context creation.
+    /// Without quota key new_session_context->setUser(*user_id, true) can throw exception
```

Comment:
(kind of unrelated but due to l. 391, it should be safe to remove the check on l. 411)

Reply:
I'm not sure that removing 100% valid, but at the moment I see no reasons to keep it. So let's do it.

I think this check was added intentionally. Based on different issues related to Session_log. But at the moment I did not meet any of issues reported in the past. 

Session log issues general discussion starts here:
https://github.com/ClickHouse/ClickHouse/issues/51760